### PR TITLE
Add water droplet hat for arachnids

### PR DESCRIPTION
**Pull request by SlamBamActionman**
## About the PR

This PR includes a new ReagentEffect `WearableReaction` that spawns an item on the entity if possible when triggered.
While this can be used in the future for more wacky effect, it is used in this PR to spawn a Water Droplet hat for Arachnids. 

The `water droplet` spawns in an Arachnid's headslot if they get splashed with 1u of water. After this it behaves as both a hat and as drinkable food, and is able to be shared with other players. The water droplet can also be thrown, splashing into a puddle if hitting an object. Fits on Hamlet.

## Why / Balance

[TW: close-up photo of adorable spider](https://static.boredpanda.com/blog/wp-content/uuuploads/jumping-spider-waterdrop-hats-uda-dennie/jumping-spider-waterdrop-hats-uda-dennie-5.jpg)

## Technical details

`WearableReaction` checks if the indicated slot of an entity triggering it is free, and then subtracts the required value of the reagent to spawn an item in that slot.

To keep this specifically to Arachnids, it's added as a reagent reaction in the species `arachnid.yml`, similar to Diona weedkiller weaknesses.

Water Droplet uses behavior borrowed from eggs - as `DrinkComponent` doesn't use up the container its attached to, it utilizes a modified `FoodComponent` instead.

## Media

Video link:
[Youtube video link](https://www.youtube.com/watch?v=CNcXBDhHCOw)

Image:
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Arachnids slipping in water now get a cute hat!

### CODE REVIEW

**Comment by SlamBamActionman** [comment]
Welp, this is what I get for pushing before testing fully. Whoops. 

I'm not entirely certain of what is causing the failure - it appears to be related to audio of gibbed creatures? Not sure how that's where it is going wrong.

**Comment by ItsMeThom** [comment]
This is one of the most rediculous things Ive ever seen. I absolutely love it.

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using Content.Shared.Inventory;
+using Content.Shared.Chemistry.Reagent;
+using JetBrains.Annotations;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chemistry.ReagentEffects
+{
```
Comment (TheShuEd) [inline Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:7]: don't use parentheses here.  look at examples of more modern systems and components to see how they are designed.

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+using Content.Shared.Inventory;
+using Content.Shared.Chemistry.Reagent;
+using JetBrains.Annotations;
+using Robust.Shared.Prototypes;
```
Comment (TheShuEd) [inline Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:4]: Jetbrains?

**Review by TheShuEd — CHANGES_REQUESTED** [review]
Some minor edits

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:4]: remove this and funny annotation

Comment (SlamBamActionman) [reply Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:4]: Fairly certain the JetBrains stuff came from whatever ReagentEffect I used as a base for this one. Just leftover crap, should be removed.

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs (hunk 1 above)
Comment (SlamBamActionman) [reply Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:6]: I assume you mean that I'm using multiple parantheses? I've cleaned it up to look nicer, should be similar in look to other ReagentEffect components now.

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+using Content.Shared.Inventory;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Chemistry.ReagentEffects
```
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:5]: Please switch to [file-scoped namespaces](https://docs.spacestation14.com/en/general-development/codebase-info/conventions.html#file-layout).

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public sealed partial class WearableReaction : ReagentEffect
```
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:7]: Above this, please add some brief documentation explaining what this does and what it's for. You can incorporate some text from the PR description.

File: Content.Server/Chemistry/ReagentEffects/WearableReaction.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    {
+
```
Comment (Partmedia) [inline Content.Server/Chemistry/ReagentEffects/WearableReaction.cs:9]: This extraneous space here can be removed.

**Review by Partmedia — CHANGES_REQUESTED** [review]
Thanks for your PR. I've marked a few nitpicks for you to take a look at.

While this reagent effect seems a little specific, I can't think of a better way to handle this right now and think this is fine.

It also seems a bit strange to me that this becomes a hat that you can take on and off, but that also seems fine to me. Is this intended only for arachnids and hamlets?

**Comment by SlamBamActionman** [comment]
> Thanks for your PR. I've marked a few nitpicks for you to take a look at.
> 
> While this reagent effect seems a little specific, I can't think of a better way to handle this right now and think this is fine.
> 
> It also seems a bit strange to me that this becomes a hat that you can take on and off, but that also seems fine to me. Is this intended only for arachnids and hamlets?

Thanks for the review!

The effect is only triggerable on arachnids, however once created may be given to other species to wear/drink. The initial concept had it break upon removal, but being able to remove it allows sharing (a fun cross-species moment!) and is a nice homage to [raindrop cakes](https://en.wikipedia.org/wiki/Raindrop_cake).

**Comment by CrigCrag** [comment]
this is the best pr i've ever seen

**Review by Partmedia — APPROVED** [review]

**Comment by Partmedia** [comment]
I'm dropping the price to 1 speso to reduce the contrived but possible situation where players repeatedly create water droplet hats and make a ton of money.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Remove the pet hat.

**Pull request merged**